kedro_airflow_k8s/cli.py

```diff
@@ -1,13 +1,79 @@
 from collections import defaultdict
 from pathlib import Path
+from typing import Callable
 
 import click
 import jinja2
+from jinja2.environment import TemplateStream
 from slugify import slugify
 
 from kedro_airflow_k8s.context_helper import ContextHelper
 
 
+def _create_template_stream(context_helper) -> TemplateStream:
+    loader = jinja2.FileSystemLoader(str(Path(__file__).parent))
+    jinja_env = jinja2.Environment(
+        autoescape=True, loader=loader, lstrip_blocks=True
+    )
+    jinja_env.filters["slugify"] = slugify
+    template = jinja_env.get_template("airflow_dag_template.j2")
+
+    package_name = context_helper.context.package_name
+
+    pipeline = context_helper.context.pipelines.get("__default__")
+    dependencies = defaultdict(list)
+
+    nodes_with_no_deps = set(node.name for node in pipeline.nodes)
+    for node, parent_nodes in pipeline.node_dependencies.items():
+        for parent in parent_nodes:
+            dependencies[parent].append(node)
+            nodes_with_no_deps = nodes_with_no_deps - set([node.name])
+
+    all_parent_nodes = set()
+    for _, parent_nodes in pipeline.node_dependencies.items():
+        all_parent_nodes = all_parent_nodes.union(
+            set(parent.name for parent in parent_nodes)
+        )
+    bottom_nodes = set(node.name for node in pipeline.nodes) - all_parent_nodes
+
+    return template.stream(
+        dag_name=package_name,
+        dependencies=dependencies,
+        project_name=context_helper.project_name,
+        pipeline=pipeline,
+        config=context_helper.config,
+        git_info=context_helper.session.store["git"],
+        base_nodes=nodes_with_no_deps,
+        bottom_nodes=bottom_nodes,
+        mlflow_url=context_helper.mlflow_config["mlflow_tracking_uri"],
+        env=context_helper.env,
+        include_start_mlflow_experiment_operator=(
+            Path(__file__).parent / "operators/start_mlflow_experiment.py"
+        ).read_text(),
+    )
+
+
+def _process_file(path: str, processor: Callable):
```

Comment:
Not sure if this needs to be so complicated, have a look on how I use fsspec [here](https://gitlab.com/getindata/ml-ops/beam-scoring/-/blob/master/beam_scoring/predict.py#L33). It can recognise filesystem out of the box.

Reply:
Great, I was looking for such functionality, and didn't realize it's that simple.